programs/wen-new-standard/src/utils.rs

```diff
@@ -53,3 +58,38 @@ pub fn get_extra_meta_list_account_pda(mint: Pubkey) -> Pubkey {
 pub fn get_approve_account_pda(mint: Pubkey) -> Pubkey {
     Pubkey::find_program_address(&[APPROVE_ACCOUNT_SEED, mint.as_ref()], &crate::id()).0
 }
+
+/// Returns true if the transfer is a cpi call
+/// currently executing `Transaction`
+pub fn is_cpi(instruction_sysvar_account_info: &AccountInfo) -> Result<bool> {
```

Comment:
You don't need the `instruction_sysvar` account to make a CPI check. You can use the following syscall:
```rust
/// Determine if we are in CPI
pub fn in_cpi() -> bool {
    get_stack_height() > TRANSACTION_LEVEL_STACK_HEIGHT
}
```
Reference: https://docs.rs/solana-program/latest/solana_program/instruction/fn.get_stack_height.html